tests/bootstrap.php

```diff
@@ -18,3 +19,15 @@ public function __construct($url)
 		$this->status_code = 200;
 	}
 }
+
+/**
+ * Acts as a fake feed request that redirects to a different URL
+ */
+class MockSimplePie_RedirectingFile extends MockSimplePie_File
+{
+	public function __construct($url)
+	{
+		parent::__construct($url);
+		$this->url = 'https://example.com/new-feed/';
```

Comment:
I do not really like hardcoding this in the Mock. The redirect URI should be set in test cases that require it but I do not see a way too do it since SimplePie requires mocking for even a simple things like this, rather than using DI, as is standard these days.

One possibility would be using anonymous classes from PHP 7, but we still need to support PHP 5.6, since it is maintained by Debian until June 2020.

* https://wiki.debian.org/PHP#Available_versions
* https://wiki.debian.org/LTS/Jessie

Reply:
could you just append a character to the url rather than replacing the whole string?

Reply:
I just made it to change the URI scheme. It will make the difference harder to notice but at least it is not hard-coding stuff.